pkg/agents/team4/finalAgent/treaty.go

```diff
@@ -0,0 +1,210 @@
+		fallthrough
+	case treaty.ConditionOp() == messages.GT && a.CurrPlatFood() > food.FoodType(treaty.ConditionValue()):
+		fallthrough
+	case treaty.ConditionOp() == messages.GE && a.CurrPlatFood() >= food.FoodType(treaty.ConditionValue()):
+		a.updateFoodFromTreatyToAgent(treaty)
+	}
+}
+
+func (a *CustomAgentEvo) handleActiveTreatyConditions() {
+	for _, treaty := range a.ActiveTreaties() {
+		if treaty.SignatureCount() > 1 {
+			if treaty.Condition() == messages.HP {
+				a.handleActiveTreatyConditionsHP(treaty)
+			} else if treaty.Condition() == messages.Floor {
+				a.handleActiveTreatyConditionsFloor(treaty)
+			} else if treaty.Condition() == messages.AvailableFood {
+				a.handleActiveTreatyConditionsAvailableFood(treaty)
+			}
+		}
+	}
+}
+
+/*------------CHANGE THE AMOUNT OF FOOD WE TAKE DEPENDING ON THE ACTIVE TREATY CONDITIONS ------------*/
+
+//--------UpdateFoodFromTreatyToAgent explanation--------//
+
+//messages.LeaveFoodAmt GT- check if a.currPlatFood - intended < request : foodtaken = a.currPlatFood - request - 1
+//GE - foodtaken = a.currPlatFood - request
+//LT - foodtaken = a.currPlatFood - request + 1
+//LE - foodtaken = a.currPlatFood - request
+//EQ - foodtaken = a.currPlatFood - request
+
+//messages.LeavePercentFood GT - check if a.currPlatFood - intended < a.currentPlatFood * foodpercent/100 : foodtaken = a.currPlatFood - a.currplatfood * request/100  - 1
+//GE - foodtaken = a.currPlatFood - (request/100 * a.currentPlatFood)
+//LT - foodtaken = a.currPlatFood - (request/100 * a.currentPlatFood) + 1
+//LE - foodtaken = a.currPlatFood - (request/100 * a.currentPlatFood)
+//EQ - foodtaken = a.currPlatFood - (request/100 * a.currentPlatFood)
+
+func (a *CustomAgentEvo) updateFoodFromTreatyToAgent(treaty messages.Treaty) {
+	switch {
+	case treaty.Request() == messages.LeaveAmountFood:
+		if treaty.RequestOp() == messages.EQ {
+			a.params.intendedFoodToTake = a.CurrPlatFood() - food.FoodType(treaty.RequestValue())
+		} else if treaty.RequestOp() == messages.GT && a.CurrPlatFood()-a.params.intendedFoodToTake < food.FoodType(treaty.RequestValue()) {
+			a.params.intendedFoodToTake = food.FoodType(math.Max(0, float64(a.CurrPlatFood()-food.FoodType(treaty.RequestValue())-1)))
+
+		} else if treaty.RequestOp() == messages.GE && a.CurrPlatFood()-a.params.intendedFoodToTake <= food.FoodType(treaty.RequestValue()) {
+			a.params.intendedFoodToTake = food.FoodType(math.Max(0, float64(a.CurrPlatFood()-food.FoodType(treaty.RequestValue()))))
+
+		} else if treaty.RequestOp() == messages.LT && a.CurrPlatFood()-a.params.intendedFoodToTake > food.FoodType(treaty.RequestValue()) {
+			a.params.intendedFoodToTake = food.FoodType(math.Max(0, float64(a.CurrPlatFood()-food.FoodType(treaty.RequestValue())+1)))
+
+		} else if treaty.RequestOp() == messages.LE && a.CurrPlatFood()-a.params.intendedFoodToTake >= food.FoodType(treaty.RequestValue()) {
```

Comment:
Convert to case and combine with GE condition?

Reply:
they have different conditions in the rest of the statement